rust/agama-cli/src/questions.rs

```diff
@@ -62,53 +62,48 @@ pub enum Modes {
     NonInteractive,
 }
 
-async fn set_mode(proxy: QuestionsProxy<'_>, value: Modes) -> anyhow::Result<()> {
-    proxy
-        .set_interactive(value == Modes::Interactive)
-        .await
-        .map_err(|e| e.into())
+async fn set_mode(client: HTTPClient, value: Modes) -> anyhow::Result<()> {
+    let policy = if value == Modes::Interactive {
```

Comment:
here it would be better to have match to ensure that newly added modes are covered correctly...and even better would be to have enum Modes implement into Policy.

Reply:
I agree, I will adapt the code.